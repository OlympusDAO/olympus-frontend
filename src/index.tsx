// import { ApolloClient, ApolloProvider, InMemoryCache } from "@apollo/client";

// import React from "react";
import ReactDOM from "react-dom";
import Root from "./Root";

// const subgraphUri = "http://localhost:8000/subgraphs/name/scaffold-eth/your-contract";

// const client = new ApolloClient({
//   uri: subgraphUri,
//   cache: new InMemoryCache(),
// });

ReactDOM.render(<Root />, document.getElementById("root"));
