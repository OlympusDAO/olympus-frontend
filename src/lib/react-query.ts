import { QueryClient } from "react-query";

export const queryClient = new QueryClient({
  defaultOptions: {
    queries: {
      refetchOnMount: false,
      refetchOnWindowFocus: false,
      notifyOnChangeProps: "tracked",
    },
  },
});
