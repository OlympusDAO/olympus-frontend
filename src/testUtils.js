import React from "react";
import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import { BrowserRouter } from "react-router-dom";
import { ThemeProvider } from "@material-ui/core/styles";
import CssBaseline from "@material-ui/core/CssBaseline";
import { light as lightTheme } from "./themes/light.js";
import { Web3ContextProvider } from "./hooks/web3Context";
import store from "./store.ts";
import { createMemoryHistory } from "history";
import App from "./App";
import { I18nProvider } from "@lingui/react";
import { i18n } from "@lingui/core";

const AllTheProviders = ({ children }) => {
  return (
    <Web3ContextProvider>
      <Provider store={store}>
        <BrowserRouter basename={"/"}>
          <ThemeProvider theme={lightTheme}>
            <CssBaseline />
            {children}
          </ThemeProvider>
        </BrowserRouter>
      </Provider>
    </Web3ContextProvider>
  );
};

const customRender = (ui, options) => render(ui, { wrapper: AllTheProviders, ...options });

const renderRoute = function (route) {
  const history = createMemoryHistory();
  history.push(route);
  render(
    <Web3ContextProvider>
      <Provider store={store}>
        <I18nProvider i18n={i18n}>
          <BrowserRouter basename={"/#"}>
            <App />
          </BrowserRouter>
        </I18nProvider>
      </Provider>
    </Web3ContextProvider>,
  );
};

// re-export everything
export * from "@testing-library/react";

// override render method
export { customRender as render, renderRoute };
