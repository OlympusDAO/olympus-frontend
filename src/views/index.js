export { default as Bond } from "./Bond/Bond";
export { default as ChooseBond } from "./ChooseBond/ChooseBond";
export { default as Stake } from "./Stake/Stake";
export { default as Wrap } from "./Wrap/Wrap";
export { default as Dashboard } from "./Dashboard/Dashboard";
export { default as PoolTogether } from "./33Together/33Together.jsx";
export { default as TreasuryDashboard } from "./TreasuryDashboard/TreasuryDashboard";
