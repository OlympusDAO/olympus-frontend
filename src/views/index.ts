export { default as Bond } from "./Bond/Bond";
export { default as ChooseBond } from "./ChooseBond/ChooseBond";
export { default as Stake } from "./Stake/Stake";
export { default as V1Stake } from "./V1-Stake/V1-Stake";
export { default as PoolTogether } from "./33Together/33together";
export { default as Wrap } from "./Wrap/Wrap";
export { default as TreasuryDashboard } from "./TreasuryDashboard/TreasuryDashboard";
export { default as Zap } from "./Zap/Zap";
