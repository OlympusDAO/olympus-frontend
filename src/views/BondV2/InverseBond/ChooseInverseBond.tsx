import { Trans } from "@lingui/macro";
import {
  Box,
  Grid,
  Table,
  TableBody,
  TableCell,
  TableContainer,
  TableHead,
  TableRow,
  Typography,
  useMediaQuery,
} from "@material-ui/core";
import { useHistory } from "react-router-dom";
import { useAppSelector, useWeb3Context } from "src/hooks";
import { usePathForNetwork } from "src/hooks/usePathForNetwork";

import { BondDataCard, BondTableData } from "../BondRow";

function ChooseInverseBond() {
  const { networkId } = useWeb3Context();
  const history = useHistory();
  usePathForNetwork({ pathName: "bonds", networkID: networkId, history });

  const inverseBonds = useAppSelector(state => {
    return state.inverseBonds.indexes
      .map(index => state.inverseBonds.bonds[index])
      .sort((a, b) => b.discount - a.discount);
  });

  const isBondsLoading = useAppSelector(state => state.inverseBonds.loading ?? true);

  const isSmallScreen = useMediaQuery("(max-width: 733px)"); // change to breakpoint query

  return (
    <>
      {inverseBonds.length === 0 && !isBondsLoading && (
        <Box display="flex" justifyContent="center" marginY="24px">
          <Typography variant="h4">
            <Trans>No active bonds</Trans>
          </Typography>
        </Box>
      )}
      {!isSmallScreen && inverseBonds.length != 0 && (
        <Grid container item>
          <TableContainer>
            <Table aria-label="Available bonds">
              <TableHead>
                <TableRow>
                  <TableCell align="center">
                    <Trans>Bond</Trans>
                  </TableCell>
                  <TableCell align="left">
                    <Trans>Payout Asset</Trans>
                  </TableCell>
                  <TableCell align="left">
                    <Trans>Price</Trans>
                  </TableCell>
                  <TableCell align="left">
                    <Trans>Discount</Trans>
                  </TableCell>
                </TableRow>
              </TableHead>
              <TableBody>
                {inverseBonds.map(bond => {
                  if (bond.displayName !== "unknown")
                    return <BondTableData networkId={networkId} key={bond.index} bond={bond} inverseBond={true} />;
                })}
              </TableBody>
            </Table>
          </TableContainer>
        </Grid>
      )}
      <Box mt={2} mb={isSmallScreen && 1} className="help-text">
        <em>
          <Typography variant="body2">
            <Trans>
              Important: Inverse bonds allow you to bond your OHM for treasury assets. Vesting time is 0 and payouts are
              instant.
            </Trans>
          </Typography>
        </em>
      </Box>
      {/* standard bonds for mobile, desktop is above */}
      {isSmallScreen && (
        <Box className="ohm-card-container">
          <Grid container item spacing={2}>
            {inverseBonds.map(bond => {
              return (
                <Grid item xs={12} key={bond.index}>
                  <BondDataCard key={bond.index} bond={bond} networkId={networkId} inverseBond={true} />
                </Grid>
              );
            })}
          </Grid>
        </Box>
      )}
    </>
  );
}
export default ChooseInverseBond;
