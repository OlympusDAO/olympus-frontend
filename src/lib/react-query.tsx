import React from "react";
import { QueryCache, QueryClient, QueryClientProvider } from "react-query";
import { ReactQueryDevtools } from "react-query/devtools";
import { Environment } from "src/helpers/environment/Environment/Environment";

const queryCache = new QueryCache({
  onError: (error, query) => {
    if (error instanceof Error) console.error({ key: query.queryKey, error: error.message });
  },
});

const queryClient = new QueryClient({
  queryCache,
  defaultOptions: {
    queries: {
      refetchOnMount: false,
      refetchInterval: false,
      refetchOnReconnect: false,
      refetchOnWindowFocus: false,
      notifyOnChangeProps: "tracked",
    },
  },
});

export const ReactQueryProvider: React.FC = ({ children }) => (
  <QueryClientProvider client={queryClient}>
    {Environment.env.NODE_ENV === "development" && <ReactQueryDevtools />}

    {children}
  </QueryClientProvider>
);
