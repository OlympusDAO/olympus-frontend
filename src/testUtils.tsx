import { i18n } from "@lingui/core";
import { I18nProvider } from "@lingui/react";
import CssBaseline from "@material-ui/core/CssBaseline";
import { ThemeProvider } from "@material-ui/core/styles";
import { render, RenderOptions, RenderResult } from "@testing-library/react";
import { createMemoryHistory } from "history";
import React, { ReactElement, ReactNode } from "react";
import { Provider } from "react-redux";
import { BrowserRouter } from "react-router-dom";

import App from "./App";
import { Web3ContextProvider } from "./hooks/web3Context";
import { ReactQueryProvider } from "./lib/react-query";
import defaultStore from "./store";
import { light as lightTheme } from "./themes/light.js";

const customRender = (ui: ReactElement, store = defaultStore, options?: RenderOptions): RenderResult => {
  const ProviderWrapper = ({ children }: { children?: ReactNode }) => (
    <Web3ContextProvider>
      <ReactQueryProvider>
        <Provider store={store}>
          <I18nProvider i18n={i18n}>
            <BrowserRouter basename={"/"}>
              <ThemeProvider theme={lightTheme}>
                <CssBaseline />
                {children}
              </ThemeProvider>
            </BrowserRouter>
          </I18nProvider>
        </Provider>
      </ReactQueryProvider>
    </Web3ContextProvider>
  );
  return render(ui, { wrapper: ProviderWrapper, ...options });
};

const renderRoute = function (route: string, store = defaultStore) {
  const history = createMemoryHistory();
  history.push(route);
  return render(
    <Web3ContextProvider>
      <ReactQueryProvider>
        <Provider store={store}>
          <I18nProvider i18n={i18n}>
            <BrowserRouter basename={"/"}>
              <App />
            </BrowserRouter>
          </I18nProvider>
        </Provider>
      </ReactQueryProvider>
    </Web3ContextProvider>,
  );
};

// re-export everything
export * from "@testing-library/react";

// override render method
export { customRender as render, renderRoute };
