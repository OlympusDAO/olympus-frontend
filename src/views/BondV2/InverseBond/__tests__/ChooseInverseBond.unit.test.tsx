import { ReactQueryProvider } from "src/lib/react-query";
import { render } from "src/testUtils";

import ChooseInverseBond from "../ChooseInverseBond";

describe("<ChooseInverseBond/>", () => {
  it("should render component", () => {
    const { container } = render(
      <ReactQueryProvider>
        <ChooseInverseBond />
      </ReactQueryProvider>,
    );
    expect(container).toMatchSnapshot();
  });
});
