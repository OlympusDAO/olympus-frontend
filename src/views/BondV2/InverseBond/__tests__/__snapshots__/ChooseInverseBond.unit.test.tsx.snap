// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ChooseInverseBond/> should render component 1`] = `
<div>
  <div
    class="MuiBox-root MuiBox-root-1"
  >
    <h4
      class="MuiTypography-root MuiTypography-h4"
    >
      No active bonds
    </h4>
  </div>
  <div
    class="MuiBox-root MuiBox-root-2 help-text"
  >
    <em>
      <p
        class="MuiTypography-root MuiTypography-body2"
      >
        Important: Inverse bonds allow you to bond your OHM for treasury assets. Vesting time is 0 and payouts are instant.
      </p>
    </em>
  </div>
</div>
`;
